Add Tarantool page (#909)
captn3m0/endoflife.date: products/tarantool.md

---
permalink: /tarantool
layout: post
title: Tarantool
command: $ tarantool --version
releasePolicyLink: https://www.tarantool.io/en/doc/latest/release/policy/
changelogTemplate: |
  https://www.tarantool.io/en/doc/latest/release/__LATEST__/

> category: db

Reviewer: Hides the missing icon:

```suggestion
category: db
iconSlug: NA
```

Author: Fixed

```diff
--- a/products/tarantool.md
+++ b/products/tarantool.md
@@ -7,6 +7,7 @@ releasePolicyLink: https://www.tarantool.io/en/doc/latest/release/policy/
 changelogTemplate: |
   https://www.tarantool.io/en/doc/latest/release/__LATEST__/
 category: db
+iconSlug: NA
 eolColumn: Support Status
 releaseDateColumn: false
 sortReleasesBy: 'releaseCycle'
```

eolColumn: Support Status
releaseDateColumn: false
sortReleasesBy: 'releaseCycle'
releases:
  - releaseCycle: "2.8"
    release: 2021-12-22
    eol: false
    latest: "2.8.3"

  - releaseCycle: "2.7"
    release: 2021-08-19
    eol: 2021-08-19
    latest: "2.7.3"

  - releaseCycle: "2.6"
    release: 2021-04-21
    eol: 2021-04-21
    latest: "2.6.3"

  - releaseCycle: "2.5"
    release: 2020-12-30
    eol: 2020-12-30
    latest: "2.5.3"

  - releaseCycle: "2.4"
    release: 2020-10-22
    eol: 2020-10-22
    latest: "2.4.3"

  - releaseCycle: "2.3"
    release: 2020-07-17
    eol: 2020-07-17
    latest: "2.3.3"

  - releaseCycle: "2.2"
    release: 2019-12-31
    eol: 2020-04-20
    latest: "2.2.2"

  - releaseCycle: "1.10"
    release: 2021-12-22
    eol: false

>     latest: "1.10.12"

Reviewer: Going by the text below, this is an LTS release. Please mark the current LTS as well.

```suggestion
    latest: "1.10.12"
    lts: true
```

Author: Fixed/

---

> > [Tarantool](https://www.tarantool.io/) is an in-memory computing platform consisting of a database and an application server. It is distributed under BSD 2-Clause terms. Key features of the application server: heavily optimized Lua interpreter with incredibly fast tracing JIT compiler, based on LuaJIT 2.1, cooperative multitasking and non-blocking IO, persistent queues, sharding, cluster and application management framework, access to external databases such as MySQL and PostgreSQL, a rich set of built-in and standalone modules.

Reviewer: Somewhat shorter:

```suggestion
> [Tarantool](https://www.tarantool.io/) is an in-memory computing platform consisting of a database and an application server. It is distributed under BSD 2-Clause terms. It includes a heavily optimized Lua interpreter, cooperative multitasking and non-blocking IO, persistent queues, sharding, and MySQL/PostgreSQL support.
```

Author: Updated:

```diff
-> [Tarantool](https://www.tarantool.io/) is an in-memory computing platform consisting of a database and an application server. It is distributed under BSD 2-Clause terms. Key features of the application server: heavily optimized Lua interpreter with incredibly fast tracing JIT compiler, based on LuaJIT 2.1, cooperative multitasking and non-blocking IO, persistent queues, sharding, cluster and application management framework, access to external databases such as MySQL and PostgreSQL, a rich set of built-in and standalone modules.
+> [Tarantool](https://www.tarantool.io/) is an open source (BSD-2 clause licensed), in-memory computing platform consisting of a database and an application server. It includes a heavily optimized Lua interpreter, cooperative multitasking and non-blocking IO, persistent queues, sharding, and MySQL/PostgreSQL support.
 
```

> The Tarantool release policy is changing to become more clear and intuitive. The new policy uses a [SemVer-like](https://semver.org/) versioning format, and introduces a new version lifecycle with more long-time support series. This document explains the new release policy, versioning rules, and [release series](https://www.tarantool.io/en/doc/latest/release/policy/#term-Release-series) lifecycle.

Reviewer: I'm not sure if talking about the existence of 2 release policies is helpful or confusing, without providing important details about the differences between these policies (such as what releases are stable and LTS, and how long the support periods are).

Reviewer 2: Thank you for comments!

In the old policy we had X.Y series with X.Y.1 as beta, X.Y.2 as stable release and X.Y.3 as the second and the last stable release. We release a series roughly once per quarter, so the support period was too short: 3 months since the first stable till the last release.

LTS is the exception here: we don't stop on X.Y.3, but continue roughly per quarter releases.

Aside of other changes the new release policy holds our intention to support a series for at least two years since a first release. We don't use the 'LTS' term in the new policy, but in fact all series become LTS.

More in-depth document about the new release policy is [here](https://github.com/tarantool/tarantool/discussions/6182).

----

To be honest, I don't know how to present the information better. We have no releases by the new policy ATM (only two beta versions) and it may be a source of a confusion. However we'll have 2.10.0 this year and, it seems, a summary of differences would be helpful.

Can (and should?) we add pre-releases (2.10.0-beta1 and 2.10.0-beta2) into the table to make things more clear? We can remove them after a release.

If you have other thoughts and/or suggestion how to better present the information, I would be very grateful.

Reviewer: Let me try to file a PR against this with some suggestions 👍🏽

Reviewer 2: Okay. Let me know about your variant of wording. I'm one of authors of the new policy and would like to see how to better present it and would like to check facts.

Author: > Let me try to file a PR against this with some suggestions 👍🏽

@captn3m0 can we merge this one if you are going to file a new PR with suggestions?

Author: @captn3m0 ?

Reviewer: Merging this, will file a separate PR when I can, will tag you for a review there. Is the https://www.tarantool.io/en/doc/latest/release/legacy-policy/ maintained on GitHub? I could file a PR there as well perhaps?

Reviewer 2: > Is the https://www.tarantool.io/en/doc/latest/release/legacy-policy/ maintained on GitHub? I could file a PR there as well perhaps?

Yep. The source is here: https://github.com/tarantool/doc/blob/latest/doc/release/legacy-policy.rst.

> ---

Reviewer 2: Is it worth to add a `releaseImage` property (I found it in the [contributing guide](https://endoflife.date/contribute)) with the image from https://www.tarantool.io/en/doc/latest/release/release-calendar/ ?

Author: Added. (It was not clear for me what does "release image" mean)

permalink: /tarantool
layout: post
title: Tarantool
command: $ tarantool --version
releasePolicyLink: https://www.tarantool.io/en/doc/latest/release/policy/
changelogTemplate: |

>   https://www.tarantool.io/en/doc/latest/release/__LATEST__/

Reviewer 2: Sadly, the release notes for 1.10.12 and 2.8.3 are still not on the website. However they're available on GitHub. At least now it would be better to point the links to GitHub.

Author: Done

Reviewer 2: But this template does not work for other releases too. Say, for 2.8.2. While GitHub releases have perfectly preditable URLs.

Reviewer 2: * 1.10.2: https://www.tarantool.io/en/doc/latest/release/1.10/#whats-new-1102
* 1.10.10: https://www.tarantool.io/en/doc/latest/release/1.10.10/
* 1.10.11: https://www.tarantool.io/en/doc/latest/release/2021-08-releases/

Author: updated tempate:

```diff
--- a/products/tarantool.md                                                 
+++ b/products/tarantool.md
@@ -6,7 +6,7 @@ command: $ tarantool --version              
 releaseImage: https://www.tarantool.io/en/doc/latest/_images/releases_calendar.svg
 releasePolicyLink: https://www.tarantool.io/en/doc/latest/release/policy/
 changelogTemplate: |                                                       
-  https://www.tarantool.io/en/doc/latest/release/__LATEST__/
+  https://github.com/tarantool/tarantool/releases/tag/__LATEST__
 category: db                      
 iconSlug: NA                                                               
 eolColumn: Support Status  
```

category: db
iconSlug: NA
eolColumn: Support Status
releaseDateColumn: false
sortReleasesBy: 'releaseCycle'
releases:
  - releaseCycle: "2.8"
    release: 2021-12-22
    eol: false
    latest: "2.8.3"

  - releaseCycle: "2.7"
    release: 2021-08-19
    eol: 2021-08-19
    latest: "2.7.3"

  - releaseCycle: "2.6"
    release: 2021-04-21
    eol: 2021-04-21
    latest: "2.6.3"

  - releaseCycle: "2.5"
    release: 2020-12-30
    eol: 2020-12-30
    latest: "2.5.3"

  - releaseCycle: "2.4"
    release: 2020-10-22
    eol: 2020-10-22
    latest: "2.4.3"

  - releaseCycle: "2.3"
    release: 2020-07-17
    eol: 2020-07-17
    latest: "2.3.3"

  - releaseCycle: "2.2"
    release: 2019-12-31
    eol: 2020-04-20

>     latest: "2.2.2"

Reviewer 2: 2.2.3 is the latest release in the 2.2 series.

Author: Fixed.

```diff
--- a/products/tarantool.md
+++ b/products/tarantool.md
@@ -16,6 +16,7 @@ releases:
     release: 2021-12-22
     eol: false
     latest: "2.8.3"
+    link: https://github.com/tarantool/tarantool/releases/tag/2.8.3
 
   - releaseCycle: "2.7"
     release: 2021-08-19
@@ -52,6 +53,7 @@ releases:
     eol: false
     lts: true
     latest: "1.10.12"
+    link: https://github.com/tarantool/tarantool/releases/tag/1.10.12
 
 ---
 
```

Reviewer 2: Still 2.2.2 here.

Author: ```diff
--- a/products/tarantool.md
+++ b/products/tarantool.md
@@ -47,7 +47,7 @@ releases:
   - releaseCycle: "2.2"
     release: 2019-12-31
     eol: 2020-04-20
-    latest: "2.2.2"
+    latest: "2.2.3"
 
   - releaseCycle: "1.10"
     release: 2021-12-22
```

  - releaseCycle: "1.10"
    release: 2021-12-22
    eol: false
    lts: true
    latest: "1.10.12"

---

> [Tarantool](https://www.tarantool.io/) is an open source (BSD-2 clause licensed), in-memory computing platform consisting of a database and an application server. It includes a heavily optimized Lua interpreter, cooperative multitasking and non-blocking IO, persistent queues, sharding, and MySQL/PostgreSQL support.

The Tarantool release policy is changing to become more clear and intuitive. The new policy uses a [SemVer-like](https://semver.org/) versioning format, and introduces a new version lifecycle with more long-time support series. This document explains the new release policy, versioning rules, and [release series](https://www.tarantool.io/en/doc/latest/release/policy/#term-Release-series) lifecycle.

The new release policy replaces the [legacy policy](https://www.tarantool.io/en/doc/latest/release/legacy-policy/) for:

- The `2.x.y` series since the `2.10.0` release. Development for this new release starts with version `2.10.0-beta1`.

> - The future `3.0.0` series.

Reviewer 2: I would say it shorter:

> ...for `2.x.y` series since `2.10.0` release and all future series (`3.x.y` and so on).

Maybe we have too much emphasis on this fact now.

Author: Done.

```diff
@@ -59,10 +62,7 @@ releases:
 
 The Tarantool release policy is changing to become more clear and intuitive. The new policy uses a [SemVer-like](https://semver.org/) versioning format, and introduces a new version lifecycle with more long-time support series. This document explains the new release policy, versioning rules, and [release series](https://www.tarantool.io/en/doc/latest/release/policy/#term-Release-series) lifecycle.
 
-The new release policy replaces the [legacy policy](https://www.tarantool.io/en/doc/latest/release/legacy-policy/) for:
-
-- The `2.x.y` series since the `2.10.0` release. Development for this new release starts with version `2.10.0-beta1`.
-- The future `3.0.0` series.
+The new release policy replaces the [legacy policy](https://www.tarantool.io/en/doc/latest/release/legacy-policy/) for 2.x.y series since 2.10.0 release and all future series (3.x.y and so on).
 
 Here are the most significant changes from the legacy release policy:
 
```
